packages/diagrams/backend/sirius-components-collaborative-diagrams/src/main/java/org/eclipse/sirius/components/collaborative/diagrams/DiagramChangeKind.java

```diff
@@ -12,16 +12,25 @@
  *******************************************************************************/
 package org.eclipse.sirius.components.collaborative.diagrams;
 
+import org.eclipse.sirius.components.annotations.Immutable;
+
 /**
  * Description of the changes performed by diagrams event handlers.
  *
  * @author fbarbin
  */
+@Immutable
 public final class DiagramChangeKind {
 
     public static final String DIAGRAM_LAYOUT_CHANGE = "DIAGRAM_LAYOUT_CHANGE"; //$NON-NLS-1$
 
+    public static final String DIAGRAM_ELEMENT_VISIBILITY_CHANGE = "DIAGRAM_ELEMENT_VISIBILITY_CHANGE"; //$NON-NLS-1$
+
     private DiagramChangeKind() {
         // Prevent instantiation
     }
+
+    public static boolean shouldCauseRefresh(String changeKind) {
+        return DIAGRAM_ELEMENT_VISIBILITY_CHANGE.equals(changeKind) || DIAGRAM_LAYOUT_CHANGE.equals(changeKind);
+    }
```

Comment:
- Do not use static methods unless it's a simple builder
- Do not change code you don't need to change (that method of one line is only used once, it's not an improvement)